File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    ls = []
    for i in range(l, r+1):
        s = str(i)
        det = 1
        for c in s:
            if c not in "50":
                det = 0
                break
        if det == 1:
            ls.append(i)
    if ls:
        return ls
    else:
        return [-1]

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int l, int r) {
    vector<int> answer;
    
    for (int i = l; i <= r; i++) {
        string num_str = to_string(i);
        bool is_valid = true;
        
        for (char ch : num_str) {
            if (ch != '0' && ch != '5') {
                is_valid = false;
                break;
            }
        }
        
        if (is_valid) {
            answer.push_back(i);
        }
    }
    
    if (answer.empty()) {
        answer.push_back(-1);
    }
    
    return answer;
}